daemon/containerd/image_exporter.go

```diff
@@ -342,3 +378,61 @@ func (i *ImageService) LoadImage(ctx context.Context, inTar io.ReadCloser, platf
 
 	return nil
 }
+
+// verifyImagesProvidePlatform checks if the requested platform is loaded.
+// If the requested platform is not loaded, it returns an error.
+func (i *ImageService) verifyImagesProvidePlatform(ctx context.Context, imgs []containerdimages.Image, platform ocispec.Platform, pm platforms.Matcher) error {
+	var missingImages []string
+	for _, img := range imgs {
+		hasRequestedPlatform := false
+		err := i.walkImageManifests(ctx, img, func(platformImg *ImageManifest) error {
+			imgPlat, err := platformImg.ImagePlatform(ctx)
+			if err != nil {
+				if cerrdefs.IsNotFound(err) {
+					return nil
+				}
+				return errors.Wrapf(err, "failed to determine image platform")
+			}
+
+			if !pm.Match(imgPlat) {
+				return nil
+			}
+			available, err := platformImg.CheckContentAvailable(ctx)
+			if err != nil {
+				return errors.Wrapf(err,
+					"failed to determine image content availability for platform %s",
+					platforms.FormatAll(platform),
+				)
+			}
+
+			if available {
+				hasRequestedPlatform = true
+				return nil
+			}
+			return nil
+		})
+		if err != nil {
+			return errdefs.System(err)
+		}
+		if !hasRequestedPlatform {
+			missingImages = append(missingImages, imageFamiliarName(img))
+		}
+	}
+
+	allImageNames := strings.Join(missingImages, ", ")
+	platformStr := platforms.FormatAll(platform)
+
+	msg := ""
+	switch len(missingImages) {
+	case 0:
+		// Success - All images provide the requested platform.
+		return nil
```

Comment:
As you're switching to an early return, I guess it could be before the `allImageNames` part above 😅